lib/signaling/v2/peerconnection.js

```diff
@@ -959,7 +959,9 @@ class PeerConnectionV2 extends StateMachine {
     }
     this._log.warn('An ICE restart has been scheduled');
     this._isIceRestartBackoffInProgress = true;
-    this._iceRestartBackoff.backoff();
+    setTimeout(() => {
```

Comment:
It does not look like new backoff is implementing the functionality that is assumed here. 

here `setTimeout` is going to fire no matter what. We need this timer to get cancelled when `this._iceRestartBackoff.reset();` is called.  The new implementation of reset does not do such thing.

Reply:
I see, I actually added a `clearTimeout` yesterday right where we called `reset` just to try some things out but removed it. Do you think that would be the proper step to take to clean up the `setTimeout`?

Reply:
Yes we do need `clearTimeout` somewhere. Basically we need it be cleared when backoff.reset is called

I suggest that you start with some unit tests against the existing Backoff module, One of the test should execute this code path that  clears the timer, Once you have those test in place, you can replace Backoff with your implementation and ensure that the tests pass again.

Reply:
Sounds good. I think it will look a bit different since our current back off module has a different API compared to our old one. Previously the timer clears the timers whereas this backoff module simply generates the numbers to be passed into a `setTimeout`. But I'll create a test on another branch with the old backoff module, and a test on this pr with similar numbers to ensure that they're indeed generating similar numbers.